src/qibolab/platform/platform.py

```diff
@@ -214,22 +177,49 @@ def disconnect(self):
                 instrument.disconnect()
         self.is_connected = False
 
-    def _execute(self, sequence, options, **kwargs):
+    def _apply_config_updates(
+        self, updates: list[dict[str, Config]]
+    ) -> dict[str, Config]:
+        """Apply given list of config updates to the default configuration and
+        return the updated config dict.
+
+        Args:
+            updates: list of updates, where each entry is a dict mapping component name to new config. Later entries
+                     in the list override earlier entries (if they happen to update the same thing).
+        """
+        components = self.component_configs.copy()
+        for update in updates:
+            for name, cfg in update.items():
+                if name not in components:
+                    raise ValueError(
+                        f"Cannot update configuration for unknown component {name}"
+                    )
+                if type(cfg) is not type(components[name]):
+                    raise ValueError(
+                        f"Configuration of component {name} with type {type(components[name])} cannot be updated with configuration of type {type(cfg)}"
+                    )
+                components[name] = cfg
+        return components
+
+    def _execute(self, sequences, options, **kwargs):
         """Executes sequence on the controllers."""
         result = {}
 
```

Comment:
I think we are missing the
```py
configs = self._apply_config_updates(options.component_configs)
```
and also
```py
        for name, cfg in configs.items():
            if name in self.instruments:
                self.instruments[name].setup(**asdict(cfg))
```
 here to have the same behavior with the `sweep`. However #861 drops the different execution methods, so if we merge it first, it should be sufficient to update one place.

Reply:
Now that I have rebased the branch on top 0.2 where there is only one execute function, this is automatically resolved.